Add document for multi nodes cluster (#2996)
replicatedhq/kURL: testgrid/README.md

1. Change directories to the root of the project
1. Run `okteto pipeline deploy -f testgrid/okteto-pipeline.yaml`
1. To "queue" a run `./bin/tgrun queue --os-spec hack/os-spec.yaml --spec hack/test-spec.yaml --ref ethan-1 --api-token this-is-super-secret --api-endpoint https://tgapi-${OKTETO_NAMESPACE}.replicated.okteto.dev`

> # Support for multi node cluster

Reviewer: ```suggestion
# Support for multi node clusters
```

> - TESTGRID is now supporting multi node cluster before it run the test
> - You can achieve that by putting the following lines in specs file

Reviewer: ```suggestion
Testgrid supports multi node clusters.
You can achieve this by putting the following lines in specs file, adjusting the number of nodes as desired.
```

> ```

Reviewer: ```suggestion
```yaml
```

  numPrimaryNodes: 1
  numSecondaryNodes: 1
```

> - This will make the cluster run with one primary node and one secondary node. 

Reviewer: ```suggestion
The previous example will make the cluster run with one primary node and one secondary node. 
```

> - The process will be divided into three stages

Reviewer: ```suggestion
The process will be divided into three stages:
```

>   1- Main process before the actual node run
> 
>   2- Initial primary node process
> 
>   3- Normal nodes process (other primary and secondary nodes)  

Reviewer: ```suggestion
1. Main process before the actual node run
1. Initial primary node process
1. Normal nodes process (other primary and secondary nodes)  
```

> ## Mainprocess

Reviewer: ```suggestion
## Main process
```

> - The following chart explain what happens in the main process

Reviewer: ```suggestion
The following chart explains what happens in the main process:
```

![flowchart-mainprocess](./assets/flowchart-mainprocess.png)

## InitPrimary node

> - The primary node can be in the following status
> 
>    1- ``running`` when the VMI start
> 
>    2- ``JoincommandStored`` when it does store the join command
> 
>    3- ``Success`` when the test run successfully 
> 
>    4- ``Failed`` when any steps during the test fail

Reviewer: ```suggestion
The primary node can be in any one of the following statuses:

1. `running` when the VMI start
1. `JoincommandStored` when it does store the join command
1. `Success` when the test run successfully 
1. `Failed` when any steps during the test fail
```

> - The following chart explain what happens in the initial primary node

Reviewer: ```suggestion
The following chart explains what happens in the initial primary node:
```

![flowchart-initialprimarynode](./assets/flowchart-InitialPrimaryNode.png)

> ## Other Nodes Flowchart

Reviewer: ```suggestion
## Additional Nodes Flowchart
```

> - Normal Nodes can be in the following status
> 
>    1- ``Running`` when the VMI strat
> 
>    2- ``WaitingJoinCommand`` when the waiting process start
> 
>    3- ``Joined`` When the node join successfully to the cluster
> 
>    4- ``Success`` when the test run successfully and the initial primary node finish the test
> 
>    5- ``failed`` when it fails to run any step
>    
> - The following chart explains what happens in the normal node (secondary and normal primary nodes)

Reviewer: ```suggestion
Additional nodes can be in the following status:

1. `Running` when the VMI starts
1. `WaitingJoinCommand` when waiting on the process to start
1. `Joined` When the node is joined successfully to the cluster
1. `Success` when the test runs successfully and the initial primary node finishes the test
1. `failed` when it fails to run any step
   
The following chart explains what happens in the normal node (secondary and normal primary nodes):
```
